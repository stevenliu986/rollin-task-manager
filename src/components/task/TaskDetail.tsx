import { Flex, Typography } from "antd";
import type { ITask } from "./AddTask";

const { Title, Paragraph } = Typography;

interface ITaskDetailProps {
    taskDetail: ITask,
}

const TaskDetail: React.FC<ITaskDetailProps> = ({ taskDetail }) => {
    const TITLE_STYLE = {
        margin: 0,
        color: '#FFF',
        fontSize: '1rem',
        fontWeight: 600,
        flex: 1,
        textDecoration: taskDetail.complete ? 'line-through' : 'none',
        opacity: taskDetail.complete ? 0.7 : 1
    };

    const PARAGRAPH_STYLE = {
        marginBottom: '24px',
        color: '#FFF',
        fontSize: '0.875rem',
        fontWeight: 400,
        textDecoration: taskDetail.complete ? 'line-through' : 'none',
        opacity: taskDetail.complete ? 0.7 : 1,
    }

    return (<Flex
        vertical
        justify='center'
        style={{
            width: '230px',
            height: '63px',
        }}
    >
        <Title
            level={4}
            style={TITLE_STYLE}
        >
            {taskDetail.title}
        </Title>
        <Paragraph
            type="secondary"
            style={PARAGRAPH_STYLE}
        >
            {taskDetail.description}
        </Paragraph>
    </Flex>)
}

export default TaskDetail;
