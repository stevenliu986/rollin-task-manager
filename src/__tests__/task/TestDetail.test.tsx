import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import TaskDetail from '../../components/task/TaskDetail';
import { Flex } from 'antd';
import type { ITask } from '../../components/task/AddTask';

vi.mock('antd', async (importOriginal) => {
    const mod = await importOriginal();
    return {
        ...mod,
        Typography: {
            Title: ({ children, style }: any) => (
                <div data-testid="task-title" style={style}>{children}</div>
            ),
            Paragraph: ({ children, style }: any) => (
                <div data-testid="task-description" style={style}>{children}</div>
            )
        }
    };
});

describe('TaskDetail Component', () => {
    const mockTask: ITask = {
        id: 1,
        title: 'Test Task',
        description: 'Test Description',
        complete: false,
        created: '2023-01-01'
    };

    const mockCompletedTask: ITask = {
        ...mockTask,
        complete: true
    };

    it('renders task title and description correctly', () => {
        render(
            <Flex>
                <TaskDetail taskDetail={mockTask} />
            </Flex>
        );

        expect(screen.getByTestId('task-title')).toHaveTextContent('Test Task');
        expect(screen.getByTestId('task-description')).toHaveTextContent('Test Description');
    });

    it('matches snapshot for incomplete task', () => {
        const { asFragment } = render(
            <Flex>
                <TaskDetail taskDetail={mockTask} />
            </Flex>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('matches snapshot for completed task', () => {
        const { asFragment } = render(
            <Flex>
                <TaskDetail taskDetail={mockCompletedTask} />
            </Flex>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});