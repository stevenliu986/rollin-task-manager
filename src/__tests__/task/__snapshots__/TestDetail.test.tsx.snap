// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TaskDetail Component > matches snapshot for completed task 1`] = `
<DocumentFragment>
  <div
    class="ant-flex css-dev-only-do-not-override-vrrzze"
  >
    <div
      class="ant-flex css-dev-only-do-not-override-vrrzze ant-flex-align-stretch ant-flex-justify-center ant-flex-vertical"
      style="width: 230px; height: 63px;"
    >
      <div
        data-testid="task-title"
        style="margin: 0px; color: rgb(255, 255, 255); font-size: 1rem; font-weight: 600; flex: 1; text-decoration: line-through; opacity: 0.7;"
      >
        Test Task
      </div>
      <div
        data-testid="task-description"
        style="margin-bottom: 24px; color: rgb(255, 255, 255); font-size: 0.875rem; font-weight: 400; text-decoration: line-through; opacity: 0.7;"
      >
        Test Description
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TaskDetail Component > matches snapshot for incomplete task 1`] = `
<DocumentFragment>
  <div
    class="ant-flex css-dev-only-do-not-override-vrrzze"
  >
    <div
      class="ant-flex css-dev-only-do-not-override-vrrzze ant-flex-align-stretch ant-flex-justify-center ant-flex-vertical"
      style="width: 230px; height: 63px;"
    >
      <div
        data-testid="task-title"
        style="margin: 0px; color: rgb(255, 255, 255); font-size: 1rem; font-weight: 600; flex: 1; text-decoration: none; opacity: 1;"
      >
        Test Task
      </div>
      <div
        data-testid="task-description"
        style="margin-bottom: 24px; color: rgb(255, 255, 255); font-size: 0.875rem; font-weight: 400; text-decoration: none; opacity: 1;"
      >
        Test Description
      </div>
    </div>
  </div>
</DocumentFragment>
`;
